index.js

```diff
@@ -119,14 +111,46 @@
   //  $$type :: String
   var $$type = '@@type';
 
-  //  type :: Any -> String
+  //  RPARSE :: RegExp
+  var RPARSE = /^(?:([^/\n\r]+)\/)?(.+?)(?:@(\d+))?$/;
+
+  //       TypeIdentifier :: (a, b, c) -> TypeIdentifier a b c
+  function TypeIdentifier(namespace, name, version) {
+    return {
+      namespace: namespace,
+      name: name,
+      version: version
+    };
+  }
+
+  //       parseTypeIdentifier :: String
+  //                           -> TypeIdentifier String? String Number?
+  function parseTypeIdentifier(s) {
+    var parsed = RPARSE.exec(s);
+    // parsed is null if the string was empty, or contained newlines
```

Comment:
Your function behaves very different on the edges:

input | `RegExp` | `indexOf`
--------|-------------|--------------
`/Type` | `null, '/Type', null` | `null, 'Type', null`
`@me` | `null, '@me', null` | `null, '', 'me'`
`///` | `null, '///', null` | `'', '//', null`
`a/b@2@1` | `'a', 'b@2', 1` | `'a', 'b', '2@1'`

Most notably it: 

* Doesn't guarantee a non-empty namespace
* Doesn't guarantee a numeric version
* Splits at the first `@` rather than the last (easy fix with `lastIndexOf`)
* Strips away `/` and `@` even if the input did not conform to the specified format

Maybe you consider none of those to be invalid behavior, in which case we can change it. I wanted the return-object to be as consistent as possible, and the parser to touch as few inputs as possible, which is why I opted for the regular expression.